==18474== Helgrind, a thread error detector
==18474== Copyright (C) 2007-2017, and GNU GPL'd, by OpenWorks LLP et al.
==18474== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==18474== Command: ./bin/kernel.out /home/utnso/git/tp-2022-1c-recargadOS/kernel/config/kernel.cfg
==18474== Parent PID: 18473
==18474== 
==18474== ---Thread-Announcement------------------------------------------
==18474== 
==18474== Thread #1 is the program's root thread
==18474== 
==18474== ---Thread-Announcement------------------------------------------
==18474== 
==18474== Thread #2 was created
==18474==    at 0x49B0152: clone (clone.S:71)
==18474==    by 0x48752EB: create_thread (createthread.c:101)
==18474==    by 0x4876E0F: pthread_create@@GLIBC_2.2.5 (pthread_create.c:817)
==18474==    by 0x4842917: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x10AF5F: inicializarPlanificacion (planificador.c:201)
==18474==    by 0x10B8A7: main (main.c:69)
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during read of size 1 at 0x4A85D6A by thread #1
==18474== Locks held: none
==18474==    at 0x4844C76: strlen (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4962C63: __tzstring (tzset.c:121)
==18474==    by 0x4964C4F: __tzfile_compute (tzfile.c:708)
==18474==    by 0x4963A20: __tz_convert (tzset.c:580)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10C9E9: esperar_cliente (socket.c:32)
==18474==    by 0x10B8C2: main (main.c:75)
==18474==  Address 0x4a85d6a is 618 bytes inside a block of size 633 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4963FEA: __tzfile_read (tzfile.c:275)
==18474==    by 0x4963084: tzset_internal (tzset.c:405)
==18474==    by 0x49639FB: __tz_convert (tzset.c:577)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during read of size 1 at 0x4A85D6B by thread #1
==18474== Locks held: none
==18474==    at 0x4844C84: strlen (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4962C63: __tzstring (tzset.c:121)
==18474==    by 0x4964C4F: __tzfile_compute (tzfile.c:708)
==18474==    by 0x4963A20: __tz_convert (tzset.c:580)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10C9E9: esperar_cliente (socket.c:32)
==18474==    by 0x10B8C2: main (main.c:75)
==18474==  Address 0x4a85d6b is 619 bytes inside a block of size 633 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4963FEA: __tzfile_read (tzfile.c:275)
==18474==    by 0x4963084: tzset_internal (tzset.c:405)
==18474==    by 0x49639FB: __tz_convert (tzset.c:577)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during read of size 1 at 0x4A85650 by thread #1
==18474== Locks held: none
==18474==    at 0x48478AD: bcmp (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4962CA7: __tzstring_len (tzset.c:93)
==18474==    by 0x4962CA7: __tzstring (tzset.c:121)
==18474==    by 0x4964C4F: __tzfile_compute (tzfile.c:708)
==18474==    by 0x4963A20: __tz_convert (tzset.c:580)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10C9E9: esperar_cliente (socket.c:32)
==18474==    by 0x10B8C2: main (main.c:75)
==18474==  Address 0x4a85650 is 16 bytes inside a block of size 20 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4962D18: __tzstring_len (tzset.c:98)
==18474==    by 0x4962D18: __tzstring (tzset.c:121)
==18474==    by 0x49646B0: __tzfile_read (tzfile.c:408)
==18474==    by 0x4963084: tzset_internal (tzset.c:405)
==18474==    by 0x49639FB: __tz_convert (tzset.c:577)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during read of size 1 at 0x4A85773 by thread #1
==18474== Locks held: none
==18474==    at 0x4845820: strcmp (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4964D65: __tzfile_compute (tzfile.c:748)
==18474==    by 0x4963A20: __tz_convert (tzset.c:580)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10C9E9: esperar_cliente (socket.c:32)
==18474==    by 0x10B8C2: main (main.c:75)
==18474==  Address 0x4a85773 is 19 bytes inside a block of size 20 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4962CBC: __tzstring_len (tzset.c:98)
==18474==    by 0x4962CBC: __tzstring (tzset.c:121)
==18474==    by 0x49646B0: __tzfile_read (tzfile.c:408)
==18474==    by 0x4963084: tzset_internal (tzset.c:405)
==18474==    by 0x49639FB: __tz_convert (tzset.c:577)
==18474==    by 0x4865167: temporal_get_string_time (in /usr/lib/libcommons.so)
==18474==    by 0x48644B1: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during write of size 1 at 0x4A85EE0 by thread #1
==18474== Locks held: none
==18474==    at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10C9E9: esperar_cliente (socket.c:32)
==18474==    by 0x10B8C2: main (main.c:75)
==18474==  Address 0x4a85ee0 is 0 bytes inside a block of size 1,024 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4912D33: _IO_file_doallocate (filedoalloc.c:101)
==18474==    by 0x4922EFF: _IO_doallocbuf (genops.c:347)
==18474==    by 0x4921F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
==18474==    by 0x49206E4: _IO_new_file_xsputn (fileops.c:1244)
==18474==    by 0x49206E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during write of size 1 at 0x4A879EC by thread #1
==18474== Locks held: none
==18474==    at 0x48488A6: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x4913DBA: fputs (iofputs.c:38)
==18474==    by 0x4864DB6: txt_write_in_file (in /usr/lib/libcommons.so)
==18474==    by 0x4864534: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10B8E5: main (main.c:77)
==18474==  Address 0x4a879ec is 60 bytes inside a block of size 4,096 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4912D33: _IO_file_doallocate (filedoalloc.c:101)
==18474==    by 0x4922EFF: _IO_doallocbuf (genops.c:347)
==18474==    by 0x4921F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
==18474==    by 0x49206E4: _IO_new_file_xsputn (fileops.c:1244)
==18474==    by 0x49206E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x4913DBA: fputs (iofputs.c:38)
==18474==    by 0x4864DB6: txt_write_in_file (in /usr/lib/libcommons.so)
==18474==    by 0x4864534: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474== Possible data race during write of size 1 at 0x4A879B0 by thread #1
==18474== Locks held: none
==18474==    at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x4913DBA: fputs (iofputs.c:38)
==18474==    by 0x4864DB6: txt_write_in_file (in /usr/lib/libcommons.so)
==18474==    by 0x4864534: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A7FF: ingresarANew (planificador.c:33)
==18474==    by 0x10B73A: manejarProceso (main.c:34)
==18474==    by 0x10B8F1: main (main.c:78)
==18474==  Address 0x4a879b0 is 0 bytes inside a block of size 4,096 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4912D33: _IO_file_doallocate (filedoalloc.c:101)
==18474==    by 0x4922EFF: _IO_doallocbuf (genops.c:347)
==18474==    by 0x4921F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
==18474==    by 0x49206E4: _IO_new_file_xsputn (fileops.c:1244)
==18474==    by 0x49206E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x4913DBA: fputs (iofputs.c:38)
==18474==    by 0x4864DB6: txt_write_in_file (in /usr/lib/libcommons.so)
==18474==    by 0x4864534: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ---Thread-Announcement------------------------------------------
==18474== 
==18474== Thread #3 was created
==18474==    at 0x49B0152: clone (clone.S:71)
==18474==    by 0x48752EB: create_thread (createthread.c:101)
==18474==    by 0x4876E0F: pthread_create@@GLIBC_2.2.5 (pthread_create.c:817)
==18474==    by 0x4842917: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x10AF88: inicializarPlanificacion (planificador.c:205)
==18474==    by 0x10B8A7: main (main.c:69)
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474==  Lock at 0x110220 was first observed
==18474==    at 0x4843D9D: pthread_mutex_init (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x10B78D: main (main.c:44)
==18474==  Address 0x110220 is 0 bytes inside data symbol "mutex_estado_ready"
==18474== 
==18474== Possible data race during write of size 1 at 0x4A85EEE by thread #3
==18474== Locks held: 1, at address 0x110220
==18474==    at 0x48488A6: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x49079A1: __vfprintf_internal (vfprintf-internal.c:1373)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x10AA93: obtenerSiguienteSRT (planificador.c:105)
==18474==    by 0x10B02C: planificacionSRT (planificador.c:225)
==18474==    by 0x10AEA3: algoritmoPlanificacion (planificador.c:185)
==18474==    by 0x10A99F: readyAexec (planificador.c:72)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474== 
==18474== This conflicts with a previous write of size 1 by thread #2
==18474== Locks held: none
==18474==    at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==  Address 0x4a85eee is 14 bytes inside a block of size 1,024 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4912D33: _IO_file_doallocate (filedoalloc.c:101)
==18474==    by 0x4922EFF: _IO_doallocbuf (genops.c:347)
==18474==    by 0x4921F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
==18474==    by 0x49206E4: _IO_new_file_xsputn (fileops.c:1244)
==18474==    by 0x49206E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474==  Lock at 0x110220 was first observed
==18474==    at 0x4843D9D: pthread_mutex_init (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x10B78D: main (main.c:44)
==18474==  Address 0x110220 is 0 bytes inside data symbol "mutex_estado_ready"
==18474== 
==18474== Possible data race during write of size 1 at 0x4A85EEF by thread #3
==18474== Locks held: 1, at address 0x110220
==18474==    at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x49087DC: __vfprintf_internal (vfprintf-internal.c:1687)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x10AA93: obtenerSiguienteSRT (planificador.c:105)
==18474==    by 0x10B02C: planificacionSRT (planificador.c:225)
==18474==    by 0x10AEA3: algoritmoPlanificacion (planificador.c:185)
==18474==    by 0x10A99F: readyAexec (planificador.c:72)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474== 
==18474== This conflicts with a previous write of size 1 by thread #2
==18474== Locks held: none
==18474==    at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==  Address 0x4a85eef is 15 bytes inside a block of size 1,024 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4912D33: _IO_file_doallocate (filedoalloc.c:101)
==18474==    by 0x4922EFF: _IO_doallocbuf (genops.c:347)
==18474==    by 0x4921F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
==18474==    by 0x49206E4: _IO_new_file_xsputn (fileops.c:1244)
==18474==    by 0x49206E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== ----------------------------------------------------------------
==18474== 
==18474==  Lock at 0x110220 was first observed
==18474==    at 0x4843D9D: pthread_mutex_init (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x10B78D: main (main.c:44)
==18474==  Address 0x110220 is 0 bytes inside data symbol "mutex_estado_ready"
==18474== 
==18474== Possible data race during write of size 1 at 0x4A85EF1 by thread #3
==18474== Locks held: 1, at address 0x110220
==18474==    at 0x48488A6: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x4908015: __vfprintf_internal (vfprintf-internal.c:1719)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x10AA93: obtenerSiguienteSRT (planificador.c:105)
==18474==    by 0x10B02C: planificacionSRT (planificador.c:225)
==18474==    by 0x10AEA3: algoritmoPlanificacion (planificador.c:185)
==18474==    by 0x10A99F: readyAexec (planificador.c:72)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4876608: start_thread (pthread_create.c:477)
==18474==    by 0x49B0162: clone (clone.S:95)
==18474== 
==18474== This conflicts with a previous write of size 1 by thread #2
==18474== Locks held: none
==18474==    at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4920661: _IO_new_file_xsputn (fileops.c:1236)
==18474==    by 0x4920661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==  Address 0x4a85ef1 is 17 bytes inside a block of size 1,024 alloc'd
==18474==    at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==    by 0x4912D33: _IO_file_doallocate (filedoalloc.c:101)
==18474==    by 0x4922EFF: _IO_doallocbuf (genops.c:347)
==18474==    by 0x4921F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
==18474==    by 0x49206E4: _IO_new_file_xsputn (fileops.c:1244)
==18474==    by 0x49206E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==18474==    by 0x490812B: __vfprintf_internal (vfprintf-internal.c:1688)
==18474==    by 0x48F2D6E: printf (printf.c:33)
==18474==    by 0x4864DED: txt_write_in_stdout (in /usr/lib/libcommons.so)
==18474==    by 0x486457F: _log_write_in_level (in /usr/lib/libcommons.so)
==18474==    by 0x48641ED: log_info (in /usr/lib/libcommons.so)
==18474==    by 0x10A835: newAready (planificador.c:39)
==18474==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==18474==  Block was alloc'd by thread #2
==18474== 
==18474== 
==18474== Use --history-level=approx or =none to gain increased speed, at
==18474== the cost of reduced accuracy of conflicting-access information
==18474== For lists of detected and suppressed errors, rerun with: -s
==18474== ERROR SUMMARY: 628 errors from 10 contexts (suppressed: 371 from 112)
